crates/cranelift/src/debug/transform/simulate.rs

```diff
@@ -69,24 +69,32 @@ fn generate_line_info(
 
     for (symbol, map) in maps {
         let base_addr = map.offset;
-        out_program.begin_sequence(Some(write::Address::Symbol { symbol, addend: 0 }));
+        out_program.begin_sequence(Some(write::Address::Symbol {
+            symbol,
+            addend: base_addr as i64,
+        }));
+
+        // Always emit a row for offset zero - debuggers expect this.
```

Comment:
> debuggers expect this

At least, LLDB expects this. When LLDB stops at a certain address (say due to a step-in), its "has debug info" check is implemented as:
```
[current address].debug_line_row.IsValid()
```
So even if we have the right DWARF DIE, etc, if we don't have a sequence in the line table that matches the current address, we get behaviors associated with functions that don't have DI.